lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -89,155 +134,122 @@ class StorageEnclosureConfig(Command):
-                f"Updating controller secret in Cortx ConfStor"
-            )
-            Conf.set(
-                'node_info_index',
-                f'storage_enclosure>{enc_id}>controller>secret',
-                password
-            )
-        if controller_type is not None:
-            self.logger.debug(
-                f"Updating controller type to {controller_type} in pillar"
-            )
-            PillarSet().run(
-                f'storage/{enc_num}/controller/type',
-                f'{controller_type}',
-                local=True
-            )
-            self.logger.info(
-                f"Updating controller type to {controller_type} in Cortx ConfStor"
-            )
-            Conf.set(
-                'node_info_index',
-                f'storage_enclosure>{enc_id}>controller>type',
-                controller_type
-            )
-        if mode is not None:
-            ip = kwargs.get('ip')
-            port = kwargs.get('port')
-
-            if ip is None and port is None:
-                self.logger.exception(
-                    f"Sub options for mode {mode} are missing"
-                )
-                raise RuntimeError('Please provide the correct'
-                    ' options for mode, exiting.')
-
-            self.logger.debug(
-                f"Updating {mode} controller IP/port in pillar"
-            )
-            if ip is not None:
-                self.logger.debug(
-                    f"Updating {mode} controller IP [{ip}] in pillar"
-                )
-                PillarSet().run(
-                    f'storage/{enc_num}/controller/{mode}/ip',
-                    f'{ip}',
-                    local=True
-                )
-                self.logger.info(
-                    f"Updating {mode} controller IP [{ip}] in Cortx ConfStor"
-                )
-                Conf.set(
-                    'node_info_index',
-                    f'storage_enclosure>{enc_id}>controller>{mode}>ip',
-                    ip
-                )
-            if port is not None:
-                self.logger.debug(
-                    f"Updating {mode} controller port in pillar"
-                )
-                PillarSet().run(
-                    f'storage/{enc_num}/controller/{mode}/port',
-                    f'{port}',
-                    local=True
-                )
-                self.logger.info(
-                    f"Updating {mode} controller port in Cortx ConfStor"
-                )
-                Conf.set(
-                    'node_info_index',
-                    f'storage_enclosure>{enc_id}>controller>{mode}>port',
-                    port
-                )
+
+        if user != None and password != None and ip != None and port != None:
+            # fetch enclosure serial/id
+            enclosure_id = EnclosureInfo(ip, user, password, port).get_enclosure_serial()
+            machine_id = grains_get("machine_id")[node_id]["machine_id"]
```

Comment:
I have a feeling machine_id and enclosure_id should be class instance variables.

Reply:
Comment addressed!